core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -312,25 +312,30 @@ public int hashCode() {
         return nodeId.hashCode();
     }
 
+    private String toString;
```

Comment:
I'm not convinced that:

> we basically always use DiscoveryNode#toString anyway.

re:
>  Out of curiosity, did you look at the places in ZenDiscovery where this is used?

I scanned it , as you say, it's hard to find. All the places I saw, including the ones you mention are things that happen every once in a while, relate to a very small subset of the nodes in the specific cluster state instance. 

I would prefer not to calculate the toString all the time. I'm ok with the optimization...